CMakeLists.txt

```diff
@@ -278,7 +278,7 @@ add_library(codegen_internal OBJECT ${NVFUSER_SRCS})
 if(NOT MSVC)
   target_compile_options(codegen_internal PRIVATE
     -Wall -Wno-unused-function
-    # -Werror
```

Comment:
I forgot why we turned off this.

Reply:
The manylinux container is using gcc 12.2, which is not supported. This version of gcc have a bug that makes it unable to compile one of my PR with `-Werror`.

Reply:
Ideally, we should upgrade to gcc 12.4, but I tried that, and it is very hard compile gcc 12.4 in AlmaLinux 8....

Reply:
Maybe we should just turn on `-Werror` for the CI clang-build job

Reply:
sounds good. I'll back out from that.

Reply:
BTW, because `-Wall` is already broken on gcc 12.2, we can continue the C++20 migration plan. Because the `-Wall` was the blocker, but now, `-Wall` is disabled due to a failure in C++17, we can just continue ignoring the warnings.